lib/fluent/plugin/out_memcached.rb

```diff
@@ -33,12 +31,21 @@ def start
 
   def shutdown
     @memcached.close
+    super
   end
 
   def format(tag, time, record)
     [tag, time, record].to_msgpack
```

Comment:
> It means <buffer> or <inject> section, right?

Yes. I mean that with the following configuration, Fluentd v0.14 can contain tag information in chunk's metadata:

```aconf
<match **>
   @type memcached
   <buffer tag>
      @type memory
      # <snip>
   </buffer>
   # <snip>
</match>
```



> Could you give me an example of config or code to reduce chunk size?

The following code is an example code for reducing chunk size.

```ruby
config_section :buffer do
  config_set_default :@type, DEFAULT_BUFFER_TYPE
  # The below config enforces to contain tag information in chunk.
  config_set_default :chunk_keys, ['tag'] 
end

def configure(conf)
  super
  # <snip>
  unless @chunk_key_tag
    raise Fluent::ConfigError, "'tag' in chunk_keys is required."
  end
end

def format(tag, time, record)
  # Omit tag serialization
  [time, record].to_msgpack
end

def write(chunk)
  tag = chunk.metadata.tag
  chunk.msgpack_each do |(time, data)|
  # <snip>
end
```